scipy/signal/signaltools.py

```diff
@@ -1959,13 +1966,20 @@ def resample_poly(x, up, down, axis=0, window=('kaiser', 5.0)):
     down //= g_
     if up == down == 1:
         return x.copy()
-    n_out = (x.shape[axis] * up) // down
-
-    # Design a linear-phase low-pass FIR filter
-    max_rate = max(up, down)
-    f_c = 1. / max_rate  # cutoff of FIR filter (rel. to Nyquist)
-    half_len = 10 * max_rate  # reasonable cutoff for our sinc-like function
-    h = firwin(2 * half_len + 1, f_c, window=window)
+    n_out = int(np.ceil(x.shape[axis] * up / down))
```

Comment:
This used to effectively be `floor`, why the change to `ceil`?

I assume it's for the extra sample in case it doesn't divide evenly, In that case it would actually be better to use integer arithmetic if possible so we stay exact, like:
```
n_out = x.shape['axis'] * up
n_out = n_out // down + bool(n_out % down)
```